consensus/src/epoch_manager.rs

```diff
@@ -263,9 +251,48 @@ impl EpochManager {
                     vec![1; proposers.len()]
                 };
 
+                // First two epochs are immediatelly skipped after a few transactions, so we don't want to look at their history.
+                let epoch_to_proposers = if use_history_from_previous_epoch && epoch_state.epoch > 2
```

Comment:
Why do we need this `epoch_state.epoch > 2` check here - is it because the `get_epoch_ending_ledger_infos` DB call fails? I am wondering if there is a cleaner way to avoid the special casing this ?

Reply:
First transaction after genesis is epoch=1, round=2, and it triggers an epoch change.
Second transaction after genesis is epoch=2, round=1.

So I think it is useless to consider any statistics from epoch < 2. That means we need to take previous epoch's history only when we are in epoch 3. but if you prefer consistency I can change it to consider the single block from epoch 1 as well. But there is no epoch 0, so when I am in epoch 1, I cannot be taking previous epoch's history.

Reply:
we probably shouldn't make the assumption that first transaction will always trigger epoch change, there's a TODO to add real timestamp to genesis.